### DISCUSSION TITLE
J'aimerai aider, par quoi commencer ?

### DISCUSSION THREAD

**Discussion by Maestro31** [openwhyd/openwhyd#566] [General]
Salut @adrienjoly , 

Merci pour ce talk de qualité sur le refactoring en archi hexagonale que j'ai découvert récemment et qui m'a donné l'occasion de découvrir également ce projet.

J'aimerai aider au refactoring en archi hexa qui a été commencé et régler quelques issues par la même occasion. Je pense que ce sera l'occasion idéale de m'améliorer sur ces sujets en pratiquant sur une code base déjà bien avancée.

Par quoi souhaiterais tu que je commence ?

**Comment by adrienjoly** [comment]
Salut Emmanuel, et merci pour ton message !

Pour commencer, je t'invite à commencer par créer un compte sur openwhyd.org pour te familiariser avec l'usage du service, puis installer la codebase pour faire tourner les tests. (cf https://github.com/openwhyd/openwhyd/blob/master/docs/INSTALL.md)

Ensuite, je vois deux manières possibles de procéder:
- soit tu commences par choisir un issue qui t'intéresse, puis on vois ensemble comment refactorer le code sous-jacent pour rendre sa résolution de cet issue la plus propre et robuste possible.
- sinon, tu pourrais continuer l'hexagonalisation qu'on a débutée, en choisissant un scope voisin de celui qu'on a montré dans le talk, ex: un autre bout de la fonction `insert`.

Quelle direction te motiverait le plus ?

**Reply by Maestro31** [reply]
J'ai déjà installé le projet et créé un compte admin en local qui m'a permi d'expérimenter un peu. Toute la suite de test run bien à priori.

J'ai également un peu regardé les différentes issues. Notamment la montée de version du manifest pour l'extension que je pensais facile. En réalité, les règles de sécurités liés à la nouvelle API ne permettent plus d'installer un script en remote sans que le site cible ait une politique de sécurité adéquate. Autrement dit, plus possible d'injecter le script sans qu'il soit dans les sources de l'extension elle même. Ça peut se comprendre,  le script remote pourrait être corrompu et atteindre tous les utilisateurs.

Le côté refacto m'intéresse bien, ça me permettrait d'avoir plus rapidement une vue d'ensemble sur l'architecture du projet et identifier où peuvent être les problèmes liés à certaines issues.

**Comment by adrienjoly** [comment]
Merci pour ta réponse !

Et d’avoir regardé la question de mise à jour de l’extension Chrome. On
dirait que ça va être galère de rester sur le Chrome Web Store… Mais au
pire il restera la possibilité d’utiliser le bookmarklet, je pense.

Écoute, si le refactoring t’intéresse, il me semble que le plus simple
serait de continuer dans le périmètre qu’on a présenté dans le talk devoxx:
la fonction insert de /app/controllers/api/post.js. En effet, on avait pas
mal couvert cette partie de tests approval et intégration, au delà de ce
qui était nécessaire pour la création de playlist. (sous-partie qui a été
hexagonalisee dans le talk)

Par exemple: remplacer la fonction repost ? A moins que tu vois une autre
fonction candidate qui paraisse plus simple pour commencer ?
